zetasql/fuzzing/zetasql_expression_proto_to_string.cc

```diff
@@ -0,0 +1,146 @@
+
+inline void HandleUndefined(const std::string& error) {
+  std::cerr << error << std::endl;
+  std::abort();
+}
+
+inline std::string HandleDefault() {
+  return "0";
+}
+
+inline std::string Padded(const std::string& s) {
+  return " " + s + " ";
+}
+
+
+TO_STRING(Expression, expr) {
+  using ExprType = Expression::ExprOneofCase;
+  switch (expr.expr_oneof_case()) {
+    case ExprType::kLiteral:
+      return LiteralExprToString(expr.literal());
+    case ExprType::kExpr:
+      return CompoundExprToString(expr.expr());
+    default:
+      return HandleDefault();
+  }
+}
+
+TO_STRING(LiteralExpr, literal) {
+  using LitExprType = LiteralExpr::LiteralOneofCase;
+  switch (literal.literal_oneof_case()) {
+    case LitExprType::kSpecialLiteral:
+      switch (literal.special_literal()) {
+        case LiteralExpr::V_NULL:
+          return "NULL";
+        default:
+          HandleUndefined("Undefined Special Literal");
+      }
+    case LitExprType::kBoolLiteral:
+      return literal.bool_literal() ? "TRUE" : "FALSE";
+    case LitExprType::kBytesLiteral:
+      return literal.bytes_literal();
+    case LitExprType::kStringLiteral:
+      return literal.string_literal();
+    case LitExprType::kIntegerLiteral:
+      return IntegerLiteralToString(literal.integer_literal());
+    case LitExprType::kNumericLiteral:
+      return NumericLiteralToString(literal.numeric_literal());
+    default:
+      return HandleDefault();
+  }
+}
+
+TO_STRING(IntegerLiteral, integer) {
+  using IntergerType = IntegerLiteral::IntegerOneofCase;
+  using std::to_string;
+  switch (integer.integer_oneof_case()) {
+    case IntergerType::kInt32Literal:
+      return to_string(integer.int32_literal());
+    case IntergerType::kUint32Literal:
+      return to_string(integer.uint32_literal());
+    case IntergerType::kInt64Literal:
+      return to_string(integer.int64_literal());
+    case IntergerType::kUint64Literal:
+      return to_string(integer.uint64_literal());
+    default:
+      return HandleDefault();
+  }
+}
+
+TO_STRING(NumericLiteral, numeric) {
+  return numeric.value();
+};
+
+TO_STRING(CompoundExpr, comp_expr) {
+  using CompoundExprType = CompoundExpr::CompoundOneofCase;
+  switch (comp_expr.compound_oneof_case()) {
+    case CompoundExprType::kBinaryOperation:
+      return BinaryOperationToString(comp_expr.binary_operation());
+    default:
+      return HandleDefault();
+  }
+};
+
+inline std::string GetBinaryOperator(zetasql_expression_grammar::BinaryOperation_Operator op) {
+  switch (op) {
+  case BinaryOperation::PLUS:
+    return "+";
+  case BinaryOperation::MINUS:
+    return "-";
+  case BinaryOperation::MULTIPLY:
+    return "*";
+  case BinaryOperation::DIVIDE:
+    return "/";
+  default:
+    HandleUndefined("Undefined Binary Operator");
+  }
+}
+
+TO_STRING(BinaryOperation, binary_operation) {
+  std::string operator_str = GetBinaryOperator(binary_operation.op());
+  return ExpressionToString(binary_operation.lhs()) + 
```

Comment:
Not sure why I didn't catch this before--consider using absl::StrCat to make this a bit more efficient.